### ISSUE TITLE
[model].filter function behavior depends on criteria order

### ISSUE THREAD

**Issue by tomaszn** [wq/wq.app#82]
Filtering depends on order of provided key-value pairs; for example, my code:
```
x_id = 265;
window.app.models.inspection.filter({
       'is_recent': true, 'x_id': 265
    }).then(function(inspections) {
        console.log('A', inspections.length);
    });
window.app.models.inspection.filter({
       'x_id': 265, 'is_recent': true
    }).then(function(inspections) {
        console.log('B', inspections.length);
    });
```
displays:
```
A 1 
B 0
```

**Comment by sheppard** [comment]
Hmm, this is a bug.  There are some optimizations that only take effect for the first filter field.  They likely involve automatic string conversion that affects boolean values differently than others.

**Comment by sheppard** [comment]
This should be fixed by 9d16ac6.  If not, feel free to reopen this issue.
